src/assets/wise5/classroomMonitor/dataExport/data-export/data-export.component.ts

```diff
@@ -1190,18 +1188,127 @@ export class DataExportComponent implements OnInit {
       componentRevisionCounter,
       matchComponentState
     );
-    for (const bucket of matchComponentState.studentData.buckets) {
-      for (const item of bucket.items) {
+    if (component.choiceReuseEnabled) {
+      this.insertMatchChoiceReuseValues(row, columnNameToNumber, component, matchComponentState);
+    } else {
+      this.insertMatchDefaultValues(row, columnNameToNumber, component, matchComponentState);
+    }
+    return row;
+  }
+
+  private insertMatchDefaultValues(
+    row: string[],
+    columnNameToNumber: any,
+    component: any,
+    matchComponentState: any
+  ): void {
+    matchComponentState.studentData.buckets.forEach((bucket: Bucket) => {
+      bucket.items.forEach((item: Choice) => {
         row[columnNameToNumber[item.id]] = this.getBucketValueById(component, bucket.id);
         if (
           this.includeCorrectnessColumns &&
           this.matchService.componentHasCorrectAnswer(component)
         ) {
           this.setCorrectnessValue(row, columnNameToNumber, item);
         }
+      });
+    });
+  }
+
+  private insertMatchChoiceReuseValues(
+    row: string[],
+    columnNameToNumber: any,
+    component: any,
+    matchComponentState: any
+  ): void {
+    matchComponentState.studentData.buckets
+      .filter((bucket: Bucket) => bucket.id !== '0')
+      .forEach((bucket: Bucket) => {
+        bucket.items.forEach((item: Choice) => {
+          this.setBucketValueWithAppending(row, columnNameToNumber, component, bucket, item);
+          if (
+            this.includeCorrectnessColumns &&
+            this.matchService.componentHasCorrectAnswer(component)
+          ) {
+            this.setCorrectnessValueForChoiceReuse(row, columnNameToNumber, item);
+          }
+        });
+      });
+  }
+
+  private setBucketValueWithAppending(
```

Comment:
This "WithAppending" part of the function name is a bit confusing. How about just ```setBucketValue()```?
```suggestion
  private setBucketValue(
```